File: daq/slc/src/dqm/src/MonObject.cc
#include "MonObject.h"

#include <sstream>

using namespace Belle2;

MonObject::MonObject() throw()
  : Shape("none"), _updated(false),
    _update_id(-1), _position_id(0), _tab_id(0) {}

MonObject::MonObject(const std::string& name)
throw() : Shape(name), _updated(false),
  _update_id(-1), _position_id(0), _tab_id(0) {}

MonObject::~MonObject() throw() {}

void MonObject::readObject(Belle2::Reader& reader)
throw(Belle2::IOException)
{
  readConfig(reader);
  readContents(reader);
}

void MonObject::readConfig(Belle2::Reader& reader)
throw(Belle2::IOException)
{
  _position_id = reader.readChar();
  _tab_id = reader.readChar();
  _name = reader.readString();
}

void MonObject::writeObject(Belle2::Writer& writer)
const throw(Belle2::IOException)
{
  writeConfig(writer);
  writeContents(writer);
}

void MonObject::writeConfig(Belle2::Writer& writer)
const throw(Belle2::IOException)
{
  writer.writeChar(_position_id);
  writer.writeChar(_tab_id);
  writer.writeString(_name);
}

std::string MonObject::toString() const throw()
{
  std::stringstream ss;
  ss << "<mon-object data-type='" << getDataType() << "' "
     << "name='" << _name << "' "
     << "update-id='" << _update_id << "' "
     << "tab='" << (int) _tab_id << "' "
     << "position='" << (int) _position_id << "' />";
  return ss.str();
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/MonObject.java
package b2daq.dqm.core;

import b2daq.core.Reader;
import b2daq.core.Serializable;
import b2daq.core.Writer;
import b2daq.graphics.GShape;
import java.io.IOException;
import java.util.Date;

public abstract class MonObject extends GShape implements Serializable {

    protected boolean _updated = false;
    protected long _update_time = 0;
    protected char _position_id = 0;
    protected char _tab_id = 0;

    public void setUpdated(boolean updated) {
        _updated = updated;
    }

    public void setUpdateTime(long update_time) {
        _update_time = update_time;
    }

    public void setUpdateTime() {
        _update_time = new Date().getTime();
    }

    public void setPositionId(char position_id) {
        _position_id = position_id;
    }

    public void setTabId(char tab_id) {
        _tab_id = tab_id;
    }

    public boolean isUpdated() {
        return _updated;
    }

    public long getUpdateTime() {
        return _update_time;
    }

    public int getPositionId() {
        return _position_id;
    }

    public int getTabId() {
        return _tab_id;
    }

    abstract public String getDataType();

    public void reset() {
        _updated = false;
    }

    @Override
    public void writeObject(Writer writer) throws IOException {
        writeConfig(writer);
        writeContents(writer);
    }

    public void writeConfig(Writer writer) throws IOException {
        writer.writeChar(_position_id);
        writer.writeChar(_tab_id);
        writer.writeString(getName());
    }

    abstract public void writeContents(Writer writer) throws IOException;

    @Override
    public void readObject(Reader reader) throws IOException {
        readConfig(reader);
        readContents(reader);
    }

    public void readConfig(Reader reader) throws IOException {
        _position_id = reader.readChar();
        _tab_id = reader.readChar();
        setName(reader.readString());
    }

    abstract public void readContents(Reader reader) throws IOException;

    public String toXML() {
        String str = "<mon-object type=\"" + getDataType() + "\" name=\"" + getName()
                + "\" update_time=\"" + getUpdateTime() + "\" " + getXML() + "</mon-object>\n";
        return str;
    }

    abstract protected String getXML();

}
